kratos/processes/vtk_output_process.cpp

```diff
@@ -0,0 +1,734 @@
+        ForceBigEndian((unsigned char *)&y_coordinate);
+        outputFile.write((char *)(&y_coordinate), sizeof(float));
+        ForceBigEndian((unsigned char *)&z_coordinate);
+        outputFile.write((char *)(&z_coordinate), sizeof(float));
+    }
+
+    outputFile.close();
+}
+
+void VtkOutputProcess::WriteConditionsAndElementsBinary(ModelPart &model_part)
+{
+    std::string outputFileName = GetOutputFileName(model_part);
+    std::ofstream outputFile;
+    outputFile.open(outputFileName, std::ios::out | std::ios::app | std::ios::binary);
+
+    // write cells header
+    outputFile << "\nCELLS " << model_part.NumberOfConditions() + model_part.NumberOfElements() << " " << mVtkCellListSize << "\n";
+
+    // write elements
+    for (ModelPart::ElementIterator elem_i = model_part.ElementsBegin(); elem_i != model_part.ElementsEnd(); ++elem_i)
+    {
+
+        ModelPart::ConditionType::GeometryType &elem_geometry = elem_i->GetGeometry();
+
+        unsigned int numberOfNodes = elem_geometry.size();
+
+        ForceBigEndian((unsigned char *)&numberOfNodes);
+
+        outputFile.write((char *)(&numberOfNodes), sizeof(unsigned int));
+
+        for (unsigned int i = 0; i < elem_geometry.size(); i++)
+        {
+            int nodenum = mKratosIdToVtkId[elem_geometry[i].Id()];
+            ForceBigEndian((unsigned char *)&nodenum);
+            outputFile.write((char *)(&nodenum), sizeof(int));
+        }
+    }
+
+    // write Conditions
+    for (ModelPart::ConditionIterator condition_i = model_part.ConditionsBegin(); condition_i != model_part.ConditionsEnd(); ++condition_i)
+    {
+        ModelPart::ConditionType::GeometryType &condition_geometry = condition_i->GetGeometry();
+        unsigned int numberOfNodes = condition_geometry.size();
+
+        ForceBigEndian((unsigned char *)&numberOfNodes);
+        outputFile.write((char *)(&numberOfNodes), sizeof(unsigned int));
+
+        for (unsigned int i = 0; i < condition_geometry.size(); i++)
+        {
+
+            int nodenum = mKratosIdToVtkId[condition_geometry[i].Id()];
+            ForceBigEndian((unsigned char *)&nodenum);
+            outputFile.write((char *)(&nodenum), sizeof(int));
+        }
+    }
+
+    outputFile.close();
+}
+
+void VtkOutputProcess::WriteConditionAndElementTypesBinary(ModelPart &model_part)
+{
+    std::string outputFileName = GetOutputFileName(model_part);
+    std::ofstream outputFile;
+    outputFile.open(outputFileName, std::ios::out | std::ios::app | std::ios::binary);
+
+    // write cell types header
+    outputFile << "\nCELL_TYPES " << model_part.NumberOfConditions() + model_part.NumberOfElements() << "\n";
+
+    // write elements types
+    for (ModelPart::ElementIterator elem_i = model_part.ElementsBegin(); elem_i != model_part.ElementsEnd(); ++elem_i)
+    {
+        const unsigned int numberOfNodes = elem_i->GetGeometry().size();
+        unsigned int element_type;
+
+        if (numberOfNodes == 3)
```

Comment:
Instead of this comparisons using only the number of nodes try to use the geometries Enums. Look https://github.com/KratosMultiphysics/Kratos/blob/e3c141d95b0afa0086b3129e3a87bcd445aa248f/applications/MeshingApplication/custom_processes/mmg_process.cpp#L334